taxonomy-editor-frontend/src/pages/project/search/FiltersArea.tsx

```diff
@@ -0,0 +1,212 @@
+    };
+  };
+
+  const updateFiltersStates = useCallback((updatedFilters) => {
+    const filtersStates = initializeFilters();
+    for (const filter of updatedFilters) {
+      switch (filter.filterType) {
+        case "is":
+          switch (filter.filterValue) {
+            case "root":
+              filtersStates.isRootNodesChecked = true;
+              break;
+            case "external":
+              filtersStates.taxonomyScopeMode = "out";
+              break;
+            case "not:external":
+              filtersStates.taxonomyScopeMode = "in";
+              break;
+          }
+          break;
+        case "language":
+          if (filter.negated) {
+            filtersStates.withoutChosenLanguagesCodes.push(
+              filter.filterValue.replace("not:", "")
+            );
+          } else {
+            filtersStates.chosenLanguagesCodes.push(filter.filterValue);
+          }
+          break;
+        default:
+          break;
+      }
+    }
+    setIsRootNodesChecked(filtersStates.isRootNodesChecked);
+    setTaxonomyScope(filtersStates.taxonomyScopeMode);
+    setChosenLanguagesCodes(filtersStates.chosenLanguagesCodes);
+    setWithoutChosenLanguagesCodes(filtersStates.withoutChosenLanguagesCodes);
+  }, []);
+
+  useEffect(() => {
+    updateFiltersStates(filters);
+  }, [filters, updateFiltersStates]);
+
+  const handleCheckbox = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    filterKey: string,
+    isChecked: boolean,
+    setChecked: React.Dispatch<React.SetStateAction<boolean>>
+  ) => {
+    setCurrentPage(1);
+    if (!isChecked) {
+      setQ((prevQ) => prevQ + ` ${filterKey}`);
+    } else {
+      setQ((prevQ) => prevQ.replace(`${filterKey}`, ""));
+    }
+    setChecked(event.target.checked);
+  };
+
+  return (
+    <Box
+      sx={{
+        width: "100%",
+        m: 1,
+        borderRadius: 1,
+        bgcolor: "#f2e9e4",
+        p: 1,
+        pl: 2,
+        pr: 2,
+        display: "flex",
+        flexWrap: "wrap",
+        alignItems: "center",
+      }}
+    >
+      <FormControlLabel
+        id="root-nodes-checkbox"
+        control={
+          <Checkbox
+            sx={checkboxTheme}
+            onChange={(e) =>
+              handleCheckbox(
+                e,
+                "is:root",
+                isRootNodesChecked,
+                setIsRootNodesChecked
+              )
+            }
+            checked={isRootNodesChecked}
+          />
+        }
+        label="Root nodes"
+      />
+      <ThreeOptionsSwitch
+        filterValue={taxonomyScope}
+        setFilterValue={setTaxonomyScope}
+        options={{
+          in: { text: "IN", isNegated: true },
+          out: { text: "OUT", isNegated: false },
+        }}
+        setQ={setQ}
+        keySearchTerm="external"
+        setCurrentPage={setCurrentPage}
+      />
+      <MultipleSelectFilter
+        label="Translated into"
+        filterValue={chosenLanguagesCodes}
```

Comment:
I'm not sure why you do not pass the `setFilterValue` function to this component whereas you do that in other places

Reply:
Currently, the `filterValue` state in the `MultipleSelectFilter` component is updated based on the response from the backend. However, `MultipleSelectFilter` only updates the query (`Q`) sent to the backend. Upon reflection, I realize that both `filterValue` and `setFilterValue` from `FilterInput` should be maintained as component states rather than passed as props.